src/processing/mode/android/AndroidBuild.java

```diff
@@ -854,9 +861,72 @@ private File mkdirs(final File parent, final String name) throws SketchException
     }
     return result;
   }
+  
+  
+  private void writeMainActivity(final File srcDirectory) {
```

Comment:
@benfry 
What do you think should be the appropriate location for this file (`MainActivity.java`) ?
I would also like to point out that `build.xml` is also written by the PDE itself and is around twice the size of `MainActivity.java`. See - https://github.com/processing/processing-android/blob/master/src/processing/mode/android/AndroidBuild.java#L617